### ISSUE TITLE
Replay video and VAST

### ISSUE THREAD

**Issue by HackingAway** [theonion/videojs-vast-plugin#62]
I have a pre-roll video playing via VAST. Works well (Thank you)

After the video plays, I want to automatically reload the player and play the next video. I have tried different methods, but the pre-roll video doesn't play.

The kinda confusing part is that there is an videojs .ended() event  fires when the VAST pre-roll video ends. (I am not sure that it should from an overall perspective)

I am ok with reloading the player and getting another pre-roll video but for some reason I can't get it to try to reload another video when the ended event fires. Any ideas?

```javascript
 vid2.ready(function(){
          this.on("ended", function(){ 
			  		console.log("ended");
					var vid2 = videojs("vid2");
					vid2.dispose();
					//put it back into the same div
					videojs("vid2", {"preload":"auto","width":1920,"height":1080}, function(){
  					// Player (this) is initialized and ready.
					this.src("http://video-js.zencoder.com/oceans-clip.mp4");
					 this.ads();
			  			this.vast({
						  skip: -1,
					  url: 'https://vastserver/example'
							});
					});
					this.play();
					   
 			
          });
```

**Comment by trajkovvlatko** [comment]
Hello

Just wanted to ask if you had any progress with this problem. 
I'm stuck with the same thing and can't do it properly 
(reload the video with a different source and different pre-rolls, from a different vast file)

Thanks!

**Comment by HackingAway** [comment]
Yes, I did eventually get it to work but not in a way I would recommend. I
queried the ad server for the VAST ad, set the video source to that ad, and
then when it was done, set the source to the next video. The end event
wasn't firing reliably in firefox so on each video grab, I had to figure
out what the length of the video was in seconds and set a timer to get the
next video when that timer expired.
I was replaying video after video and the player would eventually crash so
I moved on for now. I will have to visit this again in the future.

**Comment by trajkovvlatko** [comment]
Thanks for the info.
I'm trying to avoid this way, because of the additional fetching and parsing of the vast file,
so started patching the ads and vast plugins for videojs.
Will let you know if there's any success.
Thanks again.

**Comment by lanxichan** [comment]
HI! please see this [issue](https://github.com/videojs/video.js/issues/2853), maybe will give you a little help

**Comment by trajkovvlatko** [comment]
A bit late, but here's what worked for me:

In videojs.vast.js, inside

``` javascript
Vast = function (player, settings) {
```

add this function:

``` javascript
reload: function(url){
    settings.url = url;
},
```

and when you need to trigger an ad while the video is playing, change the url with:

``` javascript
player.vast.reload(yourNewVastLink);
player.trigger("contentupdate");
```

For some browsers, you must save the last position before triggering the contentupdate and seek to that position when the ad ended. 
Hope it helps.

**Comment by lanxichan** [comment]
When playing the vast AD, the controls generally will be hidden
```
vid1.on("playing", function(){
    controls = vid1.controls();
    if (controls === true){
      vid1.one("ended", function(){
        alert("ended!");
      });
    };
  });
```
